### PR TITLE
Update : Data Structures : Arrays

### PR DESCRIPTION
**Pull request by jainaman224**
#1044

### CODE REVIEW

**Comment by koustuvsinha** [comment]
# LGTM :+1: 

Lets :package: :shipit:

File: Data-Structure-Arrays.md [hunk 1]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+
+#### Declaration of `array`
+
+```cpp
+
+int intarray[10]; // Declares an array of integer type of size 10 with elements having random values. Index ranges from 0 to 9(i.e. size-1).
+int intarray[10] = { 0 }; // Declares an array of integer of size 10 with all elements having value 0
+
+// Choose one the two declarations and then move ahead.
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 2]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
 
+
+## Arrays in C++
+
+C++ provides a data structure, `array`, which stores a fixed-size sequential collection of elements of the same data-type. An `array` is used to store a collection of data, but it is better to think of an `array` as a collection of variables of the same type.
+
+#### Declaration of `array`
+
+```cpp
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 3]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+int intarray[10] = { 0 }; // Declares an array of integer of size 10 with all elements having value 0
+
+// Choose one the two declarations and then move ahead.
+
+```
+
+#### Inserting elements to `array`:
+
+```cpp
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 4]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+intarray[0] = 1; // Inserts an integer value of 1 at index 0
+intarray[1] = 0; // Inserts an integer value of 0 at index 1
+intarray[2] = -1; // Inserts an integer value of -1 at index 2
+intarray[3] = 1; // Inserts an integer value of 1 at index 3
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 5]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+```
+
+#### Printing an `array`:
+
+```cpp
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 6]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+std::cout << intarray[0] << std::endl; // Returns 1 which is element at index of the array
+std::cout << intarray[11] << std::endl; // A random number is expected, while in reality this is `dangerous`, and is primary cause of crashes as it's accessing a memory location which does not exist.
+
+// To print all the elements of the array
+for(int i = 0; i < n; i++)
+	std::cout << intarray[i] << std::endl;
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 7]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+```
+
+#### Basic operations on `array`:
+
+```cpp
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 8]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+std::cout << sizeof(intarray)/sizeof(intarray[0]) << std::endl; // Returns the length of the array i.e. 10.
+std::cout << sizeof(intarray[0]) << std::endl; // Returns length in bytes of one array item i.e. 4 as it is an integer
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

File: Data-Structure-Arrays.md [hunk 9]
```diff
@@ -4,6 +4,58 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+```
+
+:rocket: [Run Code](https://repl.it/CWZE/3)
+
+
```
Comment (atjonathan) [inline Data-Structure-Arrays.md]: Remove this line

Comment (jainaman224) [inline Data-Structure-Arrays.md]: Which line...? Inside block that is ``` or after that...?

**Comment by alayek** [comment]
@raisedadead please check. C++ stuff :wink:

**Comment by Rafase282** [comment]
# LGTM

I didn't check the code though.

File: Data-Structure-Arrays.md [hunk 10]
```diff
@@ -4,6 +4,49 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
...
+intarray[1] = 0; // Inserts an integer value of 0 at index 1
+intarray[2] = -1; // Inserts an integer value of -1 at index 2
+intarray[3] = 1; // Inserts an integer value of 1 at index 3
+```
+
+#### Printing an `array`:
+
+```c++
+std::cout << intarray[0] << std::endl; // Returns 1 which is element at index of the array
+std::cout << intarray[11] << std::endl; // A random number is expected, while in reality this is `dangerous`, and is primary cause of crashes as it's accessing a memory location which does not exist.
```
Comment (raisedadead) [inline Data-Structure-Arrays.md]: I think you mis-interpreted [this comment](https://github.com/FreeCodeCamp/wiki/pull/1056#discussion_r65341995).

This should be:
```diff
-std::cout << intarray[11] << std::endl; // A random number is expected, while in reality this is `dangerous`, and is primary cause of crashes as it's accessing a memory location which does not exist.
+std::cout << intarray[11] << std::endl; // Would give a a "Garbage" value as there is no element at index 11 of array. That memory location is beyond the range of the array.

```

**Comment by raisedadead** [comment]
Just one slight change above and this is good to go.
Merge when addressed.

**Pull request merged**